x11-misc/gpaste/gpaste-3.36.3.ebuild

```diff
@@ -0,0 +1,64 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+VALA_MIN_API_VERSION="0.30"
+VALA_USE_DEPEND="vapigen"
+
+inherit meson vala
+
+DESCRIPTION="Clipboard management system"
+HOMEPAGE="https://github.com/Keruspe/GPaste"
+SRC_URI="https://github.com/Keruspe/GPaste/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="introspection gnome vala"
+REQUIRED_USE="vala? ( introspection )"
+
+DEPEND="
+	dev-libs/appstream-glib
+	>=dev-libs/glib-2.48:2
+	>=dev-libs/gobject-introspection-1.48.0
+	sys-apps/dbus
+	>=x11-libs/gdk-pixbuf-2.34:2
+	>=x11-libs/gtk+-3.20:3
+	x11-libs/libX11
+	x11-libs/libXi
+	gnome? (
+		gnome-base/gnome-control-center:2
+		media-libs/clutter
+		x11-libs/pango
+	)
+"
+BDEPEND="
+	vala? ( $(vala_depend) )
+	virtual/pkgconfig
+"
+RDEPEND="${DEPEND}
+	gnome? (
+		gnome-base/gnome-shell
+	)
+"
+
+S=${WORKDIR}/GPaste-${PV}
+
+src_prepare() {
+	use vala && vala_src_prepare
+	default
+}
+
+src_configure() {
+	local emesonargs=(
+		-Dsystemd=true
+		-Dbash-completion=true
+		-Dzsh-completion=true
+		$(meson_use introspection introspection)
+		$(meson_use vala vapi)
+		$(meson_use gnome x-keybinder)
```

Comment:
Regardless, it seems to require gnome-keybinder binary. I have no idea what package provides that, but if you can compile this, you probably know or at least can check it out using *equery* or alike.
```
Calculating dependencies... done!
[ebuild  N     ] x11-misc/gpaste-3.36.3::testworld  USE="-gnome -introspection -vala" 0 KiB

Total: 1 package (1 new), Size of downloads: 0 KiB

Would you like to merge these packages? [Yes/No] y

>>> Verifying ebuild manifests

>>> Emerging (1 of 1) x11-misc/gpaste-3.36.3::testworld
 * gpaste-3.36.3.tar.gz BLAKE2B SHA512 size ;-) ...                                                           [ ok ]
>>> Unpacking source...
>>> Unpacking gpaste-3.36.3.tar.gz to /var/tmp/portage/x11-misc/gpaste-3.36.3/work
>>> Source unpacked in /var/tmp/portage/x11-misc/gpaste-3.36.3/work
>>> Preparing source in /var/tmp/portage/x11-misc/gpaste-3.36.3/work/GPaste-3.36.3 ...
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/x11-misc/gpaste-3.36.3/work/GPaste-3.36.3 ...
meson setup --buildtype plain --libdir lib64 --localstatedir /var/lib --prefix /usr --sysconfdir /etc --wrap-mode nodownload --build.pkg-config-path /usr/share/pkgconfig --pkg-config-path /usr/share/pkgconfig --native-file /var/tmp/portage/x11-misc/gpaste-3.36.3/temp/meson.x86_64-pc-linux-gnu.amd64.ini -Dsystemd=true -Dbash-completion=true -Dzsh-completion=true -Dintrospection=false -Dvapi=false -Dx-keybinder=false -Dgnome-shell=false /var/tmp/portage/x11-misc/gpaste-3.36.3/work/GPaste-3.36.3 /var/tmp/portage/x11-misc/gpaste-3.36.3/work/gpaste-3.36.3-build
The Meson build system
Version: 0.55.1
Source dir: /var/tmp/portage/x11-misc/gpaste-3.36.3/work/GPaste-3.36.3
Build dir: /var/tmp/portage/x11-misc/gpaste-3.36.3/work/gpaste-3.36.3-build
Build type: native build
Project name: GPaste
Project version: 3.36.3
C compiler for the host machine: x86_64-pc-linux-gnu-gcc (gcc 10.2.0 "x86_64-pc-linux-gnu-gcc (Gentoo 10.2.0-r1 p2) 10.2.0")
C linker for the host machine: x86_64-pc-linux-gnu-gcc ld.bfd 2.34
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/x86_64-pc-linux-gnu-pkg-config (1.7.3)
Run-time dependency appstream-glib found: YES 0.7.17
Run-time dependency gdk-3.0 found: YES 3.24.22
Run-time dependency gdk-pixbuf-2.0 found: YES 2.40.0
Run-time dependency gio-2.0 found: YES 2.64.5
Run-time dependency glib-2.0 found: YES 2.64.5
Run-time dependency gobject-2.0 found: YES 2.64.5
Run-time dependency gtk+-3.0 found: YES 3.24.22
Run-time dependency pango found: YES 1.42.4
Run-time dependency dbus-1 found: YES 1.12.20
Found CMake: /usr/bin/cmake (3.18.2)
Run-time dependency gnome-keybindings found: NO (tried pkgconfig and cmake)

meson.build:34:2: ERROR: Dependency "gnome-keybindings" not found, tried pkgconfig and cmake

A full log can be found at /var/tmp/portage/x11-misc/gpaste-3.36.3/work/gpaste-3.36.3-build/meson-logs/meson-log.txt
 * ERROR: x11-misc/gpaste-3.36.3::testworld failed (configure phase):
 *   (no error message)
```

Reply:
https://github.com/Keruspe/GPaste/blob/master/meson.build#L32 and here's the option where the requirement comes from.

Reply:
I've worked out that if gnome-control-center is installed it provides the pkg-config file (gnome-keybindings.pc) so that it can find the gnome-keybindings directory (resolves to /usr/share/gnome-control-center/keybindings on my machine).

It does this regardless if gnome-shell option is enabled at compile time or not.

I'm looking for advice, as to which method to resolve:

A) Patch the meson.build file so that it only looks for gnome-keybindings when when the gnome use flag is enabled, and make gnome-control-center a build dep, 

B) Supply the meson option control-center-keybindings-dir with a path (/usr/share/gnome-control-center/keybindings) if gnome use flag is not enabled so that it doesn't attempt to find the gnome-keybindings pkg-config file

Reply:
So for Option B, that file wouldn't exist (we'd hope), then it'd fail? This wouldn't work if we wanted to disable gnome even if we had the package providing it installed. So we need to do A.

Reply:
I'm setting up a VM without all the gnome stuff to test, but at this stage it looks like gnome-control-center is a runtime dependency and is only a build dep in order to determine the keybindings directory. You can suppress it being required as a build dep by specifying the keybindings directory as a configure option.